### PR TITLE
Pem decode rest

### LINKED ISSUE #95: Failed to connect to the host via ssh: invalid format
For some reason when a terraform variable storing the ssh key is passed in via an environment variable, there are some extra bits at the end of the key by the time it reaches `terraform-provisioner-ansible`.

### Steps to reproduce

```hcl
    connection {
        type = "ssh"
        user = "ubuntu"
        private_key = "${var.ssh_private_key}"
        //private_key = "${file("~/.ssh/id_rsa")}"  WORKS
    }
```
1. `export TF_VAR_ssh_private_key="$(cat ~/.ssh/id_rsa)"`
2. `terraform apply`

### Expected behavior

Ansible provisioner executes without error

### Actual behavior

```
openstack_compute_instance_v2.infra (ansible): TASK [Gathering Facts] *********************************************************
openstack_compute_instance_v2.infra (ansible): fatal: [(ip redacted)]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Load key \"/private/var/folders/kt/8r8l44m54159f_r1k0qwsmmm0000gp/T/b660abc6-f02b-4210-b36f-72a0a3e3fea1845955416\": invalid format\r\nubuntu@(ip redacted): Permission denied (publickey).\r\n", "unreachable": true}
```

### Configuration

**Terraform version**:
`v0.11.10`

### PR DESCRIPTION
**Pull request by sjwl**
### Summary

fix #95 
use encoding/pem library to filter any extraneous bits attached to the key

### CODE REVIEW

File: mode/local_connection_extractor.go [hunk 1]
```diff
@@ -90,6 +91,11 @@ func parseConnectionInfo(s *terraform.InstanceState) (*connectionInfo, error) {
 		connInfo.TimeoutVal = DefaultTimeout
 	}
 
+	if connInfo.PrivateKey != "" {
+
```
Comment (radekg) [inline mode/local_connection_extractor.go]: Empty line.

File: mode/local_connection_extractor.go [hunk 2]
```diff
@@ -90,6 +91,11 @@ func parseConnectionInfo(s *terraform.InstanceState) (*connectionInfo, error) {
...
+		block, _ := pem.Decode([]byte(connInfo.PrivateKey))
+		connInfo.PrivateKey = string(pem.EncodeToMemory(block))
```
Comment (radekg) [inline mode/local_connection_extractor.go]: Is it possible to check here what has been decoded? The example here: https://golang.org/pkg/encoding/pem/#example_Decode suggests so. Would that be a good idea?

File: mode/local_connection_extractor.go [hunk 3]
```diff
@@ -104,6 +110,9 @@ func parseConnectionInfo(s *terraform.InstanceState) (*connectionInfo, error) {
 		}
 		if connInfo.BastionPrivateKey == "" {
 			connInfo.BastionPrivateKey = connInfo.PrivateKey
+		} else {
+			block, _ := pem.Decode([]byte(connInfo.BastionPrivateKey))
+			connInfo.BastionPrivateKey = string(pem.EncodeToMemory(block))
```
Comment (radekg) [inline mode/local_connection_extractor.go]: Is it possible to check here what has been decoded? The example here: https://golang.org/pkg/encoding/pem/#example_Decode suggests so. Would that be a good idea?

**Comment by radekg** [comment]
Hi @sjwl, thank you for the pull request. This is a very good idea. As this code is being updated, it might be a good idea to check if the decoded data is a private key?

File: mode/local_connection_extractor.go (hunk 2 above)
Comment (sjwl) [reply mode/local_connection_extractor.go]: Good idea how does this look?

File: mode/local_connection_extractor.go (hunk 3 above)
Comment (sjwl) [reply mode/local_connection_extractor.go]: Good idea how does this look?

**Comment by radekg** [comment]
The update looks really good, @sjwl. It would be great to move the decode and block type test into a separate function + add an actual test. If you feel I'm picky, I can do it after accepting the PR. Let me know what you think.

**Comment by sjwl** [comment]
@radekg ok moved the logic to a function, however I did not create a test because there is no test file `local_connection_extractor_test.go` that I could slip a small specific test for the private key.

**Comment by radekg** [comment]
@sjwl thank you for extracting the function. I am going to merge your change into an intermediate branch, add a test, merge to master and release a patch version tonight.

**Comment by radekg** [comment]
v2.1.1 released: https://github.com/radekg/terraform-provisioner-ansible/releases/tag/v2.1.1.

**Pull request merged**